### PR TITLE
spec: Fix conditional for dnf-data

### PR DESCRIPTION
**Pull request by sgallagher**
The spec for DNF 5 doesn't build properly on ELN/RHEL10. These conditionals get it building correctly, but there are test failures and I haven't figured out what's causing them yet.

### CODE REVIEW

File: dnf5.spec [hunk 1]
```diff
@@ -12,18 +12,18 @@ Source0:        %{url}/archive/%{version}/dnf5-%{version}.tar.gz
 
 Requires:       libdnf5%{?_isa} = %{version}-%{release}
 Requires:       libdnf5-cli%{?_isa} = %{version}-%{release}
-%if 0%{?fedora} <= 38
+%if (0%{?fedora} && 0%{?fedora} < 39) || (0%{?rhel} && 0%{?rhel} < 10)
```
Comment (yselkowitz) [inline dnf5.spec:15]:
```suggestion
%if ! (0%{?fedora} > 38 || 0%{?rhel} > 9)
```

File: dnf5.spec [hunk 2]
```diff
@@ -533,7 +533,7 @@ Requires:       libdnf5%{?_isa} = %{version}-%{release}
 Requires:       libdnf5-cli%{?_isa} = %{version}-%{release}
 Requires:       dbus
 Requires:       polkit
-%if 0%{?fedora} <= 38
+%if 0%{?fedora} <= 38 || 0%{?rhel} > 9
```
Comment (yselkowitz) [inline dnf5.spec:536]:
```suggestion
%if ! (0%{?fedora} > 38 || 0%{?rhel} > 9)
```

**Review by yselkowitz — CHANGES_REQUESTED** [review]

**Comment by sgallagher** [comment]
Sorry for the delay. This should be better now.

**Comment by j-mracek** [comment]
LGTM

**Pull request merged**